### ISSUE TITLE
dictionary key is converted to boolean value by formatter

### ISSUE THREAD

**Issue by klei1984** [de-jcup/eclipse-yaml-editor#85]
I am setting up a github actions script which is in yaml format. Actions has a specific keyword set which relies on `<key>: <value>` syntax. See this [page](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#on) for all the keywords.

The on.push.branches event is basically described with the below yaml code:

```
name: GitHub CI

on:
  push:
    branches:
    - master
    tags: [ '**' ]
```

I am using version 1.6.1 of the editor which converts the `on:` key to the boolean `true` value as in the below result:

```
name: GitHub CI
true:
  push:
    branches:
    - master
    tags:
    - '**'
```

I did not check what github actions actually do with such a script, but my assumption is that keys are not supposed to be replaced by boolean values.

Am I missing something? I found a similar issue [here](https://github.com/go-yaml/yaml/issues/214) but that is another tool and another context.

Or could it be that the yaml format used by Github is not actually compliant to either version of the standard and thus I should not try to use a yaml editor for such scripts?

**Comment by de-jcup** [comment]
I tried out the formatting and had some odd result as you... 

## Problem description
The problem is, that under the hood snake-yaml parser is used to load the yaml document from string content into a yaml object and after this the object is dumped back to a string. 

So we got `String -> Yaml-Object ->String`

"on" is recognized by snake-yaml as a boolean value... so inside the loaded yaml object its recognized as "boolean:true" and when doing the dump the output is switched to "true".

I have looked into the sources and found a potential solution: snake-yaml gives many possibilities to use specialized renderers and resolvers. Maybe for formatting there should customized variants which could avoid the problem.

**Comment by de-jcup** [comment]
Just released V1.6.2 on eclipse marketplace 

Formatter preference page has got a new option "prevent type conversion" which is enabled per default:

![image](https://user-images.githubusercontent.com/11644753/73461850-24766a00-437b-11ea-8754-4ee69f3ed8bb.png)

When this is enabled, your example is formatted like next screenshots:
![image](https://user-images.githubusercontent.com/11644753/73462032-6ef7e680-437b-11ea-93be-7359a912b8f0.png)
becomes:
![image](https://user-images.githubusercontent.com/11644753/73462064-7d460280-437b-11ea-92ad-78fda37acbe0.png)

**Comment by klei1984** [comment]
Thanks a lot!
